File: packages/claude-plugins/worktree/scripts/lib/paths.ts
import { resolve, dirname, basename, relative } from 'node:path'
import { existsSync, mkdirSync } from 'node:fs'

export function getWorktreesDir(mainRepoDir: string): string {
    const repoName = basename(mainRepoDir)
    return resolve(dirname(mainRepoDir), `${repoName}-worktrees`)
}

export function getConfigDir(worktreesDir: string): string {
    return resolve(worktreesDir, 'config')
}

export function getWorktreePath(worktreesDir: string, worktreeName: string): string {
    return resolve(worktreesDir, worktreeName)
}

export function getRelativePath(from: string, to: string): string {
    return relative(from, to)
}

export function ensureDir(path: string): void {
    if (!existsSync(path)) {
        mkdirSync(path, { recursive: true })
    }
}

export function worktreeNameFromBranch(branch: string): string {
    // feature/142-add-feature -> 142-add-feature
    return branch.replace(/^feature\//, '').replace(/^[^/]+\//, '')
}

export function configExists(worktreesDir: string): boolean {
    const configDir = getConfigDir(worktreesDir)
    return existsSync(configDir)
}

File: packages/towles-tool/worktree/skills/lib/paths.py
"""Path utilities for worktree management."""
from __future__ import annotations

from pathlib import Path


def get_worktrees_dir(main_repo_dir: Path) -> Path:
    """Get the worktrees sibling directory."""
    repo_name = main_repo_dir.name
    return main_repo_dir.parent / f"{repo_name}-worktrees"


def get_config_dir(worktrees_dir: Path) -> Path:
    return worktrees_dir / "config"


def get_worktree_path(worktrees_dir: Path, worktree_name: str) -> Path:
    return worktrees_dir / worktree_name


def get_relative_path(from_path: Path, to_path: Path) -> str:
    """Get relative path from one location to another."""
    try:
        return str(to_path.relative_to(from_path))
    except ValueError:
        # Not relative, use ../ style
        return str(Path("..") / to_path.relative_to(from_path.parent))


def ensure_dir(path: Path) -> None:
    path.mkdir(parents=True, exist_ok=True)


def worktree_name_from_branch(branch: str) -> str:
    """Convert branch name to worktree folder name.

    feature/142-add-feature -> 142-add-feature
    """
    import re
    return re.sub(r"^[^/]+/", "", branch)


def config_exists(worktrees_dir: Path) -> bool:
    return get_config_dir(worktrees_dir).exists()
